### ISSUE TITLE
Limit in queryParameter breaks methods

### ISSUE THREAD

**Issue by vivekspanicker** [pcrady/coinbase_exchange#2]
It looks like following error occurs when a limit variable is set.

Future<http.Response> get({
    required String path,
    Map<String, String>? headers,
    Map<String, dynamic>? queryParameters,
  }) async {
    Uri url = Uri.https(_authority, path, queryParameters);
__TypeError (type 'int' is not a subtype of type 'Iterable<dynamic>')_

**Comment by pcrady** [comment]
I'll take a look and work on more thorough tests. Thanks for finding this.

**Comment by pcrady** [comment]
Can you give an example of where you see this happening. I tried it with getHolds, getAccountLedger and getAccountTransfers and none of them had errors when I set the limit.

**Comment by vivekspanicker** [comment]
Please try getFills or GetOrders from OrdersClient. This error pops up in following method in abstract class RestClient when either of these are used.

```
Future<http.Response> get({
    required String path,
    Map<String, String>? headers,
    Map<String, dynamic>? queryParameters,
  }) async {
    **Uri url = Uri.https(_authority, path, queryParameters);**
```

**Comment by pcrady** [comment]
ah yeah this is the problem. Query parameters have to be a string or and string iterable
```
factory Uri(
    {String? scheme,
    String? userInfo,
    String? host,
    int? port,
    String? path,
    Iterable<String>? pathSegments,
    String? query,
    Map<String, dynamic /*String|Iterable<String>*/ >? queryParameters,   <---------------- HERE
    String? fragment}) = _Uri;
```
I'll go through everything and push a fix for this. Thanks for finding it.

**Comment by pcrady** [comment]
this should be resolved in version 0.0.5
